tests/chainer_tests/functions_tests/rnn_tests/test_function_slstm.py

```diff
@@ -29,146 +29,126 @@ def inject_backend_tests(method_names):
-
-        with backend_config:
-            c, h = functions.slstm(*inputs)
-            assert c.data.dtype == self.dtype
-            assert h.data.dtype == self.dtype
-
-        testing.assert_allclose(
-            c_expect, c.data, **self.check_forward_options)
-        testing.assert_allclose(
-            h_expect, h.data, **self.check_forward_options)
-
-    def test_forward(self, backend_config):
-        self.check_forward(self.inputs, backend_config)
-
-    def test_flat_forward(self, backend_config):
-        self.check_forward(self.flat(self.inputs), backend_config)
-
-    def check_backward(self, inputs, grad_outputs, backend_config):
-        if backend_config.use_cuda:
-            inputs = cuda.to_gpu(inputs)
-            grad_outputs = cuda.to_gpu(grad_outputs)
-
-        with backend_config:
-            gradient_check.check_backward(
-                functions.slstm, inputs, grad_outputs,
-                **self.check_backward_options)
-
-    def test_full_backward(self, backend_config):
-        self.check_backward(self.inputs, self.grad_outputs, backend_config)
-
-    def test_flat_full_backward(self, backend_config):
-        self.check_backward(
-            self.flat(self.inputs), self.flat(self.grad_outputs),
-            backend_config)
-
-    def test_no_gc_backward(self, backend_config):
-        grad_outputs = [None, self.grad_outputs[1]]
-        self.check_backward(self.inputs, grad_outputs, backend_config)
-
-    def test_flat_no_gc_backward(self, backend_config):
-        grad_outputs = [None, self.grad_outputs[1]]
-        self.check_backward(
-            self.flat(self.inputs), self.flat(grad_outputs), backend_config)
-
-    def test_no_gh_backward(self, backend_config):
-        grad_outputs = [self.grad_outputs[0], None]
-        self.check_backward(self.inputs, grad_outputs, backend_config)
-
-    def test_flat_no_gh_backward(self, backend_config):
-        grad_outputs = [self.grad_outputs[0], None]
-        self.check_backward(
-            self.flat(self.inputs), self.flat(grad_outputs), backend_config)
-
-    def check_double_backward(
-            self, inputs, grad_outputs, grad_grad_inputs, backend_config):
-        if backend_config.use_cuda:
-            inputs = cuda.to_gpu(inputs)
-            grad_outputs = cuda.to_gpu(grad_outputs)
-            grad_grad_inputs = cuda.to_gpu(grad_grad_inputs)
-
-        with backend_config:
-            gradient_check.check_double_backward(
-                chainer.functions.slstm, inputs, grad_outputs,
-                grad_grad_inputs, **self.check_double_backward_options)
-
-    def test_double_backward(self,  backend_config):
-        self.check_double_backward(
-            self.inputs, self.grad_outputs, self.grad_grad_inputs,
-            backend_config)
+    def generate_grad_outputs(self, outputs_template):
+        grad_out = []
+
+        h = outputs_template[0]
+        c = outputs_template[0]
```

Comment:
The return values are `c` followed by `h` so it should be `h = outputs_template[1]` I think.